tests/testthat/test-example.R

```diff
@@ -702,3 +702,52 @@ test_that("steps must not be negative", {
     mod$simulate(c(0:10, -5)),
     "All elements of 'step_end' must be non-negative")
 })
+
+
+test_that("run random walk deterministically", {
+  res <- dust_example("walk")
+  obj <- res$new(list(sd = 1), 0, 100, seed = 1L)
+  rng_state <- obj$rng_state()
+  m <- obj$state()
+  m[] <- runif(length(m))
+  obj$set_state(m)
+  expect_equal(obj$run(10, deterministic = TRUE), m)
+  expect_equal(obj$rng_state(), rng_state)
+})
+
+
+test_that("run simulate deterministically", {
+  res <- dust_example("walk")
+  obj <- res$new(list(sd = 1), 0, 100, seed = 1L)
+  rng_state <- obj$rng_state()
+  m <- obj$state()
+  m[] <- runif(length(m))
+  obj$set_state(m)
+  res <- obj$simulate(0:10, deterministic = TRUE)
+  expect_equal(res,
+               array(rep(m, 11), c(1, 100, 11)))
+  expect_equal(obj$rng_state(), rng_state)
+})
+
+
+test_that("staggered start times, deterministically", {
+  sir <- dust_example("sir")
+  mod <- sir$new(list(), 0, 10, seed = 1L)
+
+  state <- mod$state()
+  n <- round(runif(10, -5, 5))
+  state[1, ] <- state[1, ] - n
+  state[2, ] <- state[2, ] + n
+  step <- 1:10
+
+  mod$set_state(state, step, deterministic = TRUE)
+
+  res <- mod$state()
+  f <- function(i) {
+    mod <- sir$new(list(), 0, 1, seed = NULL)
+    mod$set_state(state[, i], step[i])
+    mod$run(10, deterministic = TRUE)
```

Comment:
Does the SIR example round the result from rbinom?

Reply:
no, so these non-integral numbers will persist. This is what Neil was looking for (and of course was impossible prior to changes in 0c4915c as otherwise rbinom was truncating `n`)